src/TritonCTS/src/third_party/CKMeans/clustering.cpp

```diff
@@ -244,18 +244,24 @@ float clustering::Kmeans (unsigned N, unsigned CAP, unsigned IDX, vector<pair<fl
 		
 		// flop to slot matching based on min-cost flow
         if (iter == 1) 
-            minCostFlow(means, CAP, IDX, 5200);
+            minCostFlow(means, CAP, IDX, 5200, power);
         else if (iter == 2) 
-            minCostFlow(means, CAP, IDX, 5200);
+            minCostFlow(means, CAP, IDX, 5200, power);
         else if (iter > 2) 
-            minCostFlow(means, CAP, IDX, 5200);
+            minCostFlow(means, CAP, IDX, 5200, power);
 
         // collect results
         clusters.clear();
         clusters.resize(N);
         for (unsigned i = 0; i < flops.size(); ++i) {
             flop * f = flops[i];
-            clusters[f->match_idx[IDX].first].push_back(f);
+            int position = 0;
+            if (f->match_idx[IDX].first < 0){
+                position = 0;
+            } else {
+                position = f->match_idx[IDX].first;
+            }
+            clusters[position].push_back(f);
```

Comment:
The old code had segFaults depending on the capacity/power parameters. Some clusters were set to cluster -1 and others, which were not well defined in this part of the code. I've tested and it seems setting to position 0 works best. (However, other segments of code already "try" to catch this problem. You can see a similar piece of code just after clustering on HTreeBuilder.cpp)